pkg/core/mpt/billet.go

```diff
@@ -153,8 +164,13 @@ func (b *Billet) putIntoHash(curr *HashNode, path []byte, val Node) (Node, error
 	}
 	// We also need to increment refcount in both cases. That's the only place where refcount
 	// is changed during restore process. Also flush right now, because sync process can be
-	// interrupted at any time.
+	// interrupted at any time. // TODO: flush right now?
 	b.incrementRefAndStore(val.Hash(), val.Bytes())
+
+	if val.Type() == LeafT {
```

Comment:
It can be non-leaf. And what you need to do is to go through all of its childs and increments refs for them.

Reply:
If it's not a leaf, then state sync module will handle its children and request missing ones (see `(s *Module) restoreNode(n mpt.Node)`). In the general case, we can't just iterate through children and increment refs, because children may not be fetched yet.